doc/decisions/DR025-UI-delegation-plugin-interface.md

```diff
@@ -0,0 +1,194 @@
+by unique identifier (Katana-style).
+
+The UI delegate plugin provides a factory function for instantiating a
+UI delegate object, which in turn is responsible for creating/modifying
+UI elements. The UI delegate is provided with its corresponding
+`Manager` instance on construction, thus avoiding the need for
+singletons (FnAssetAPI-style).
+
+The UI plugin system uses the same structure as the core plugin systems.
+In particular, both C++ and Python plugins can be supported.
+
+#### Pros
+
+- Trivial to choose a different language (C++/Python) for the UI plugin
+  vs. core plugin.
+- Can add Python/C++ support first then add C++/Python support later.
+- UI delegation functionality is invisible and unobtrusive if unused.
+- Allows independent development, versioning and deployment of UI
+  plugins and core plugins. A single core plugin may work across all
+  applications, whereas a UI plugin may have a different variant for
+  each framework (i.e. Qt vs. GTK vs. wxWidgets vs. FLTK ...) or even
+  each application.
+- Very few breaking changes required, if any.
+
+#### Cons
+
+- Extra boilerplate code for host applications and manager plugin
+  authors.
+- Additional deployment effort for facilities maintaining yet another
+  plugin.
+- Core and UI plugins maintained and versioned independently must
+  be kept compatible and deployed in compatible pairs.
+- The UI delegate will need access to the core manager plugin. This will
+  be limited to the public `Manager` object (like any other consumer),
+  unless the architecture of the API/plugin system is modified.
+
+### Option 2
+
+A single logical plugin using the existing plugin system, with
+additional method(s) added to the existing `ManagerInterface` class.
+I.e. a `getUIDelegate` (or similar) method returns a UI delegate object,
+which is responsible for creating/modifying UI elements.
+
+#### Pros
+
+- Choosing a different language (C++/Python) for the UI vs. core methods
+  is possible using the hybrid plugin system with the addition of a new
+  `Capability` enum value for UI related API methods.
+- Minimal additional boilerplate for host application and manager plugin
+  authors.
+- UI and core API implementations are compatible by construction.
+- The UI layer has direct access to the private implementation of the
+  core API plugin.
+
+#### Cons
+
+- Relying on the hybrid plugin system to provide multi-language support
+  adds unnecessary boilerplate to the implementation of the plugin when
+  its functionality is higher level than (and more akin to a consumer
+  of) the core API.
+- Relying on the hybrid plugin system means that support for UI
+  delegation must be added to the C++ API first. So it is not possible
+  to add Python support first and C++ support later.
+- Definitely requires breaking changes (though only additions so should
+  be source compatible).
+- Tight coupling of UI and headless concerns means care must be taken,
```

Comment:
For me, this would be a deal breaker, its critical to have a UI dependency free resolver.